pkg/connector/group.go

```diff
@@ -0,0 +1,355 @@
+package connector
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Nerzal/gocloak/v13"
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+	"github.com/conductorone/baton-sdk/pkg/pagination"
+	"github.com/conductorone/baton-sdk/pkg/types/resource"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+)
+
+type groupBuilder struct {
+	resourceType *v2.ResourceType
+	client       *Connector
+}
+
+func (o *groupBuilder) ResourceType(ctx context.Context) *v2.ResourceType {
+	return groupResourceType
+}
+
+func (o *groupBuilder) List(ctx context.Context, parentResourceID *v2.ResourceId, pToken *pagination.Token) ([]*v2.Resource, string, annotations.Annotations, error) {
+	var resources []*v2.Resource
+	annos := annotations.Annotations{}
+
+	if err := o.client.ensureConnected(ctx); err != nil {
+		return nil, "", nil, err
+	}
+
+	groups, err := o.client.client.GetGroups(ctx)
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	for _, group := range groups {
+		groupResource, err := parseIntoGroupResource(group, nil)
+		if err != nil {
+			return nil, "", nil, err
+		}
+		resources = append(resources, groupResource)
+	}
+
+	return resources, "", annos, nil
+}
+
+func (o *groupBuilder) Entitlements(ctx context.Context, resource *v2.Resource, _ *pagination.Token) ([]*v2.Entitlement, string, annotations.Annotations, error) {
+	var entitlements []*v2.Entitlement
+
+	if err := o.client.ensureConnected(ctx); err != nil {
+		return nil, "", nil, err
+	}
+
+	// Create a membership entitlement for the group
```

Comment:
We have some helpers that might be useful:

https://github.com/ConductorOne/baton-sdk/blob/22dc7aa01dabab8db9bd7069746417212e952929/pkg/types/entitlement/entitlement.go
https://github.com/ConductorOne/baton-sdk/blob/22dc7aa01dabab8db9bd7069746417212e952929/pkg/types/grant/grant.go